### PR TITLE
Add fwcfg type of sysinfo element test cases

### PR DESCRIPTION
**Pull request by meinaLi**
Add fwcfg type of sysinfo element test cases
1) Define/start guest with fw_cfg sysinfo
2) Define/start guest with invaild entry value for fw_cfg type sysinfo

Signed-off-by: Meina Li <meili@redhat.com>

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg
JOB ID     : 3cf1deed402777c2e0b3ab726232788ade339021
JOB LOG    : /root/avocado/job-results/job-2020-11-06T01.09-3cf1dee/job.log
 (1/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.positive_test.entry_value: PASS (26.33 s)
 (2/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.positive_test.entry_file: PASS (33.01 s)
 (3/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.negative_test.entry_value.without_name: PASS (29.82 s)
 (4/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.negative_test.entry_value.without_prefix_opt: PASS (28.41 s)
 (5/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.negative_test.entry_file.with_exist_name: PASS (30.78 s)
 (6/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_seabios.sysinfo_fwcfg.negative_test.entry_file.without_file: PASS (37.08 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 196.43 s
```
```
# avocado run --vt-type libvirt --vt-machine-type q35 virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg
JOB ID     : 9864316d80fa5de5689d8cc9bfdb37125b235d50
JOB LOG    : /root/avocado/job-results/job-2020-11-06T00.53-9864316/job.log
 (1/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.positive_test.entry_value: PASS (51.36 s)
 (2/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.positive_test.entry_file: PASS (62.64 s)
 (3/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.negative_test.entry_value.without_name: PASS (147.89 s)
 (4/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.negative_test.entry_value.without_prefix_opt: PASS (147.74 s)
 (5/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.negative_test.entry_file.with_exist_name: PASS (148.70 s)
 (6/6) type_specific.io-github-autotest-libvirt.virsh.boot_sysinfo.by_ovmf.sysinfo_fwcfg.negative_test.entry_file.without_file: PASS (150.71 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 719.32 s
```

**Comment by meinaLi** [comment]
This PR is related with avocado-vt: https://github.com/avocado-framework/avocado-vt/pull/2816/

File: libvirt/tests/cfg/bios/virsh_boot_sysinfo.cfg [hunk 1]
```diff
@@ -0,0 +1,44 @@
+- virsh.boot_sysinfo:
+    type = virsh_boot_sysinfo
+    start_vm = "no"
+    status_error = "no"
```
Comment (chloerh) [inline libvirt/tests/cfg/bios/virsh_boot_sysinfo.cfg:4]: Maybe putting 'status_error = no' under 'positive_test' seems better

File: libvirt/tests/src/bios/virsh_boot_sysinfo.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                sysinfo_xml.entry_name = entry_name
+            # Test with entry value in text
+            else:
+                sysinfo_xml.entry_name = entry_name
+                sysinfo_xml.entry = value_string
+        vmxml.sysinfo = sysinfo_xml
+
+        logging.debug("New VM XML is:\n%s" % vmxml)
+        ret = virsh.define(vmxml.xml)
+        logging.debug("haha:%s" % virsh.dumpxml(vm_name))
```
Comment (chloerh) [inline libvirt/tests/src/bios/virsh_boot_sysinfo.py:106]: haha ?

File: libvirt/tests/src/bios/virsh_boot_sysinfo.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            check_in_vm(test, vm, **kwargs)
+
+    finally:
+        logging.debug("Start to cleanup")
+        if vm.is_alive():
+            vm.destroy()
+        logging.debug("Restore the VM XML")
+        vmxml_backup.sync(options="--nvram")
+        if file_path:
+            process.run("rm -rf %s" % file_path, shell=True, ignore_status=True)
```
Comment (chloerh) [inline libvirt/tests/src/bios/virsh_boot_sysinfo.py:135]: Seems this 'file_path' is user-defined. Then how about putting it under  data_dir.get_tmp_dir()? if it's ok, I mean the file permissions.

**Review by chloerh — APPROVED** [review]
LGTM. BTW, it'd better if the review comments could be replied.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**